src/jitlayers.h

```diff
@@ -73,7 +73,7 @@ GlobalVariable *jl_emit_RTLD_DEFAULT_var(Module *M) JL_NOTSAFEPOINT;
 DataLayout jl_create_datalayout(TargetMachine &TM) JL_NOTSAFEPOINT;
 
 static inline bool imaging_default() JL_NOTSAFEPOINT {
-    return jl_options.image_codegen || (jl_generating_output() && jl_options.use_pkgimages);
+    return jl_options.image_codegen || jl_generating_output() || jl_options.use_pkgimages;
```

Comment:
So this seems undesirable. It effectively causes codegen to use image mode by default even for normal JIT compilation.

Reply:
The flag used to be:
```
static inline bool imaging_default() {
    return jl_options.image_codegen || (jl_generating_output() && !jl_options.incremental);
}
```

@vtjnash do we want it to be

```
jl_options.image_codegen || (jl_generating_output() && (!jl_options.incremental || jl_options.use_pkgimages))
```
???

Reply:
Sure. I’d also be okay with `return 1` here xD

Reply:
Well that would be a slightly more drastic change, but @pchintalapudi might like that as well.

We should evaluate for 1.10 if that is feasible